org.opent2t.sample.multisensor.superpopular/com.wink.sensorpod/js/thingTranslator.js

```diff
@@ -70,10 +70,15 @@ class StateReader {
 }
 
 function convertDeviceDateToTranslatorDate(unixTimestamp) {
-    // Date takes a number of milliseconds, so convert seconds to milliseconds
-    var datetime = new Date(unixTimestamp * 1000);
+    // Only convert if input is valid numeric value
+    if (unixTimestamp && !isNaN(unixTimestamp)) { 
+        // Date takes a number of milliseconds, so convert seconds to milliseconds
+        var datetime = new Date(unixTimestamp * 1000);
+        return datetime.toISOString();
+    }
 
-    return datetime.toISOString();
+    // return all other invalid input as-is since we can't convert to DateTime
+    return unixTimestamp;
```

Comment:
For this case this method should always return a DateTime or null/undefined.  It *should not* return a string.  So my opinion is we have two options, in the case where we can't create a DateTime based on *_changed_at:

1. we return null/undefined
2. we return *_updated_at

I'm ok with either, 2 *may* give us a more consistent return but also *may* have some delayed value but given this case it may be the only value we'll get.  I also recommend moving forward with option 2 given Wink's docs.

---
In reply to: [106057659](https://github.com/openT2T/translators/pull/210#discussion_r106057659) [](ancestors = 106057659)